libsolidity/interface/CompilerStack.cpp

```diff
@@ -1548,6 +1549,9 @@ string CompilerStack::createMetadata(Contract const& _contract, bool _forIR) con
 	if (m_revertStrings != RevertStrings::Default)
 		meta["settings"]["debug"]["revertStrings"] = revertStringsToString(m_revertStrings);
 
+	meta["settings"]["metadata"]["appendCBOR"] =
+		m_metadataFormat != MetadataFormat::NoMetadata;
```

Comment:
This should really only be there, if it's not set to the default, shouldn't it? That's how all (or at least most of) the others work, isn't it?

Reply:
Ok, I'll fix this.

Reply:
Yeah, this was pointed out before in one of the comments.

Personally, I ignored that concern because I'm not sure if it's a good thing. Defaults can change. But yeah, to be consistent with that we do currently, this would be omitted when it's set to the default.

Reply:
See also https://github.com/ethereum/solidity/pull/13265#discussion_r937774398

Reply:
Fixed it.